hazelcast/src/main/java/com/hazelcast/cluster/impl/ClusterServiceImpl.java

```diff
@@ -642,6 +643,16 @@ public void handleJoinRequest(JoinRequest joinRequest, Connection connection) {
                 }
             }
 
+            if (node.isMaster()) {
+                try {
+                    node.getNodeExtension().beforeJoin();
+                } catch (Exception e) {
+                    logger.warning(e.getMessage());
+                    sendBeforeJoinCheckFailure(joinRequest.getAddress(),e.getMessage());
```

Comment:
I think you have checkstyle issues here. I guess that the cluster.impl package is fully excluded from checkstyle and that is why these issues are not shown.